### PR TITLE
Extended Syntax: primops tests

### PR DESCRIPTION
**Pull request by Anabra**
See #32

### CODE REVIEW

File: grin/test/ExtendedSyntax/PrimOpsSpec.hs [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+
+
+C.include "<stdio.h>"
+-- TODO: Once, we merge the syntax tranformation PR,
+-- remove the extra ".." from here. It is needed now
+-- because of the nested directory structure.
+-- TODO: Once merged, use "../../prim_ops.c"
+C.include "../../prim_ops.h"
+
+spec :: Spec
```
Comment (luc-tielen) [inline grin/test/ExtendedSyntax/PrimOpsSpec.hs:23]: With a little more effort you could write property based tests for these primitive functions as well.

Comment (andorp) [reply grin/test/ExtendedSyntax/PrimOpsSpec.hs:23]: There are some, see below:
```
    it "works for a random string" $ monadicIO $ do
       str <- pick randomString
       run $ primStringLen str
```

Comment (andorp) [reply grin/test/ExtendedSyntax/PrimOpsSpec.hs:23]: Well at least we check if we don't do segfault :D .

Comment (andorp) [reply grin/test/ExtendedSyntax/PrimOpsSpec.hs:23]: The next thing would be looking into how a set of primops and their composition do satisfy some kind of commutative diagrams, but that is a bit overkill atm.

Comment (Anabra) [reply grin/test/ExtendedSyntax/PrimOpsSpec.hs:23]: Good point about the property-based tests. For now, let's just merge this, and address it separately.

**Pull request merged**